plugins/mimecast/komand_mimecast/tasks/monitor_siem_logs/task.py

```diff
@@ -71,31 +74,52 @@ def run(self, params={}, state={}) -> (List[Dict], Dict):  # pylint: disable=unu
                 exp = PluginException(preset=PluginException.Preset.UNKNOWN, data=gen_error)
             return [], state, has_more_pages, 500, exp
 
-    def _filter_and_sort_recent_events(self, task_output: List[Dict[str, Any]]) -> List[Dict[str, Any]]:
+    def _filter_and_sort_recent_events(self, task_output: List[Dict[str, Any]], cut_off: int) -> List[Dict[str, Any]]:
         """
         Filters and sorts a list of events to retrieve only the recent events.
 
         :param task_output: A list of dictionaries representing the task output to be filtered and sorted.
         :type: List[Dict[str, Any]]
 
+        :param cut_off: integer specifying how far back in time we filter events from
+        :type: integer
+
         :return: A new list of dictionaries representing the filtered and sorted recent events.
         :rtype: List[Dict[str, Any]]
         """
 
         self.logger.info(f"Number of raw logs returned from Mimecast: {len(task_output)}")
+        filter_time = get_time_hours_ago(hours_ago=cut_off)
         task_output = sorted(
             map(
                 lambda event: event.get_dict(),
                 filter(
-                    lambda event: event.compare_datetime(get_time_hours_ago(hours_ago=CUTOFF)),
+                    lambda event: event.compare_datetime(filter_time),
                     [EventLogs(data=event) for event in task_output],
                 ),
             ),
             key=itemgetter(EventLogs.FILTER_DATETIME),
         )
-
+        latest_time = filter_time
         for event in task_output:
+            event_date_time = event.get(EventLogs.FILTER_DATETIME)
+            latest_time = event_date_time if event_date_time > latest_time else latest_time
             event.pop(EventLogs.FILTER_DATETIME, None)
         self.logger.info(f"Number of returned logs after filtering performed: {len(task_output)}")
-
+        if task_output:
+            self.logger.info(f"Latest event time returned from Mimecast logs: {latest_time}")
         return task_output
+
+    def _get_filter_time(self, custom_config: Dict[str, int]) -> int:
+        """
+        We want to apply the custom_config params if provided to the task. The logic is that if
```

Comment:
Suggested reword: 
Apply custom_config params (if provided) to the task.
If a lookback value exists, it should take precedence (this can allow a larger filter time), otherwise use the default value.

Reply:
Updated this also.